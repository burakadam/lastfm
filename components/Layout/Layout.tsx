import Link from 'next/link';

interface ILayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: ILayoutProps) => (
  <div>
    <header className='flex justify-center py-3 border-b-2 mb-4'>
      <Link href='/' className='font-bold'>
        LASTFM
      </Link>
    </header>
    <section className='max-w-3xl m-auto px-2'>{children}</section>
  </div>
);

export default Layout;
