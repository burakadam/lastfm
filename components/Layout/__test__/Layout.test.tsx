import { render } from '@testing-library/react';

import Layout from '../Layout';

describe('<Layout> spec', () => {
  let COMP: any;

  beforeEach(() => {
    COMP = render(
      <Layout>
        <p>This is a children</p>
      </Layout>
    );
  });
  it('render layout component', () => expect(COMP).toMatchSnapshot());
});
