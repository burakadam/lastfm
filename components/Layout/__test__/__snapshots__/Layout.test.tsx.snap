// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> spec render a artist card component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="flex justify-center py-3 border-b-2 mb-4"
        >
          <a
            class="font-bold"
            href="/"
          >
            LASTFM
          </a>
        </header>
        <section
          class="max-w-3xl m-auto px-2"
        >
          <p>
            This is a children
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="flex justify-center py-3 border-b-2 mb-4"
      >
        <a
          class="font-bold"
          href="/"
        >
          LASTFM
        </a>
      </header>
      <section
        class="max-w-3xl m-auto px-2"
      >
        <p>
          This is a children
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Layout> spec render layout component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="flex justify-center py-3 border-b-2 mb-4"
        >
          <a
            class="font-bold"
            href="/"
          >
            LASTFM
          </a>
        </header>
        <section
          class="max-w-3xl m-auto px-2"
        >
          <p>
            This is a children
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="flex justify-center py-3 border-b-2 mb-4"
      >
        <a
          class="font-bold"
          href="/"
        >
          LASTFM
        </a>
      </header>
      <section
        class="max-w-3xl m-auto px-2"
      >
        <p>
          This is a children
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
